File: JAVA/All Codes/Leap_Year.java
import java.util.Scanner;
class Leap_Year 
{
public static void main(String[] args) 
{
int year;
Scanner scan = new Scanner(System.in);
System.out.println("Enter any Year:");
year = scan.nextInt();
scan.close();
 boolean isLeap = false;
if(year % 4 == 0)
{
if( year % 100 == 0)
{
 if ( year % 400 == 0)
 isLeap = true;
 else
isLeap = false;
}
else
isLeap = true;
}
else {
isLeap = false;
} 
if(isLeap==true)
System.out.println(" is a Leap Year  :"+ year );
else
System.out.println(" is not a Leap Year  :"+ year);
}
}

File: C++/All Codes/Leap_year.cpp
#include<iostream>
using namespace std;
class Leap_year
{
    int year;
    public: void check()
    {
        cout<<"Enter a year \n";
        cin>>year;
        if(year%4==0)
            {
            cout<<"  year is Leep year\n";
            }
        else{
            cout<<"year is not a leap year  \n";
            }}
};
int main()
{
    Leap_year l1;
    l1.check();
}
